packages/react-components/lib/robots/robot-table-datagrid.tsx

```diff
@@ -42,22 +42,33 @@ export function RobotDataGridTable({ onRobotClick, robots }: RobotDataGridTableP
       const working = {
         color: theme.palette.success.main,
       };
+      const disabled = {
+        color: theme.palette.grey.A400,
+      };
       const defaultColor = {
         color: theme.palette.warning.main,
       };
 
       switch (params.row.status) {
-        case Status2.Error:
+        case Status.Error:
           return error;
-        case Status2.Charging:
+        case Status.Offline:
+        case Status.Uninitialized:
+          return disabled;
+        case Status.Charging:
           return charging;
-        case Status2.Working:
+        case Status.Working:
           return working;
         default:
           return defaultColor;
       }
     })();
 
+    const robotDecommissioned =
+      params.row.commission &&
+      params.row.commission.dispatch_tasks === false &&
+      params.row.commission.direct_tasks === false;
```

Comment:
Could a robot be "partially decommissioned" where it only accepts direct tasks? (technically it could also only accept dispatch or idle tasks?). I am wondering if it would be "safer" to just consider the dispatch tasks.

Reply:
Good point. Yes it can be partially decommissioned, although our endpoint will disable both direct and dispatch tasks. But in the event that developers decommissioned only the direct tasks (for testing or whatever), this should not prevent users from recommissioning a robot.

As for our endpoint disabling both direct tasks and dispatch tasks (instead of just dispatch tasks), this is to prevent RMF from still allocating charging tasks to the robot in any situation, as those are considered direct tasks. 

I've made the change to only check the dispatch task policy to determine if the robot is decommissioned, https://github.com/open-rmf/rmf-web/pull/920/commits/fa149f05c8020e0c137f4243b490da7bc40e200f